### PR TITLE
feat(React): Search page UI improvements, 'all' entity search. 

### PR DESCRIPTION
**Pull request by jjoyce0510**
_As seen in the recent townhall!_ 

**Scope**
These changes affect the React App. 

**Changes**
1. Update SearchPage to include tabs with icons
2. Added "All" tab to search page, showing grouped entity search results. 
3. Moved the search filter experience to a modal. 
4. Add platforms icons for looker, hive, kafka, hdfs. These should be pushed from the server (coming soon). 
5. Misc style cleanups. 
6. Fixed bug in search results pagination. 

**Screenshots**
Screen Shot 2021-02-23 at 9 45 01 PM
Screen Shot 2021-02-23 at 9 45 15 PM
Screen Shot 2021-02-23 at 9 45 26 PM
Screen Shot 2021-02-23 at 9 55 31 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    const history = useHistory();
+    const entityRegistry = useEntityRegistry();
+    const { data } = useGetSearchResultsQuery({
+        variables: {
+            input: {
+                type,
+                query,
+                start: 0,
+                count: SearchCfg.RESULTS_PER_PAGE,
+                filters: null,
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:36]: should we be propagating filters here as well?

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            },
+        },
+    });
+
+    const results = data?.search?.entities || [];
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:41]: if there are no entities, should we show the List at all?

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                        >
+                            <Typography.Text>
+                                See all <b>{entityRegistry.getCollectionName(type)}</b> results
+                            </Typography.Text>
+                            <ArrowRightOutlined />
+                        </Button>
+                    )
+                }
+                style={styles.resultList}
+                dataSource={data?.search?.entities || []}
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:78]: `results`?

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    <span style={styles.header}>
+                        <Typography.Title level={3}>{entityRegistry.getCollectionName(type)}</Typography.Title>
+                        <Card bodyStyle={styles.resultHeaderCardBody} style={styles.resultHeaderCard as any}>
+                            {entityRegistry.getIcon(type, 36, IconStyleType.ACCENT)}
+                        </Card>
+                    </span>
+                }
+                footer={
+                    data?.search &&
+                    data?.search?.total > 0 && (
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:56]: should we show a `show all` button in the case that there's only one result?

Comment (gabe-lyons) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:56]: or should this be total > SearchCfg.RESULTS_PER_PAGE?

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                            <Typography.Text>
+                                See all <b>{entityRegistry.getCollectionName(type)}</b> results
+                            </Typography.Text>
+                            <ArrowRightOutlined />
+                        </Button>
+                    )
+                }
+                style={styles.resultList}
+                dataSource={data?.search?.entities || []}
+                split={false}
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:79]: why are we turning off split but adding our own dividers? whats the difference

File: datahub-web-react/src/app/search/EntitySearchResults.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    return (
+        <div style={styles.resultsContainer}>
+            {loading && <Message type="loading" content="Loading..." style={styles.loading} />}
+            <Button style={styles.addFilters} onClick={onEditFilters}>
+                <FilterOutlined />
+                Add Filters
+            </Button>
+            <Modal
+                title="Filters"
+                footer={<Button onClick={onCloseEditFilters}>Apply</Button>}
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/EntitySearchResults.tsx:88]: should we support a cancel too so they have an option to dismiss their staged changes?

File: datahub-web-react/src/app/search/SearchHeader.tsx [hunk 7]
```diff
@@ -39,43 +53,24 @@ export const SearchHeader = ({
...
-                    suggestions={suggestions}
-                    onSearch={onSearch}
-                    onQueryChange={onQueryChange}
-                />
-                <ManageAccount urn={authenticatedUserUrn} pictureLink={authenticatedUserPictureLink} />
-            </div>
+        <Header style={styles.header as any}>
+            <Link to="/">
+                <Space size={4}>
+                    {' '}
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/SearchHeader.tsx:59]: if this space is just for the sake of padding, i'd commend just using css-padding

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 1 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:36]: Ah, I see- we can't support facet filtering on all results 👍

**Review by gabe-lyons — APPROVED** [review]

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 1 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:36]: correcto!!

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:41]: it has a nice "no results" default that looks better than whitespace imo... but i get your point functionally maybe it doesn't make sense.... hmm

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 3 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:78]: oh goodness. thank you!!

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 4 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:56]: hm....... see all also serves the purpose of flipping tabs for you so you can apply filters etc.. i think see all is still okay...

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 5 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:79]: their dividers do not look like what we want.... they are full width across the list, i wanted the padding :)

File: datahub-web-react/src/app/search/EntitySearchResults.tsx (hunk 6 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntitySearchResults.tsx:88]: sure we can support the "x" button

File: datahub-web-react/src/app/search/EntityGroupSearchResults.tsx (hunk 2 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/EntityGroupSearchResults.tsx:41]: going to remove showing the list

File: datahub-web-react/src/app/search/SearchHeader.tsx (hunk 7 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/SearchHeader.tsx:59]: good catch this should not be there.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**